h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -3022,7 +3020,23 @@ public double applyBounds(double d, int i) {
       return d;
     }
 
+    private Frame encodeCategoricalsIfPresent(Frame beta_constraints) {
+      FrameUtils.BetaConstraintsEncoder constraintsEncoder = new FrameUtils.BetaConstraintsEncoder(_dinfo.coefNames(), _dinfo.coefOriginalNames());
+      Frame transformedFrame =  constraintsEncoder.doAll(getBetaConstraintsTypesForEncoder(beta_constraints), beta_constraints).outputFrame();
+      transformedFrame.setNames(beta_constraints._names);
+      return transformedFrame;
+    }
+    
+    private byte[] getBetaConstraintsTypesForEncoder(Frame beta_constraints) {
+      byte[] types = beta_constraints.types();
+      int id = Arrays.asList(beta_constraints.names()).indexOf("names");
+      types[id] = 2;
```

Comment:
what is 2? why not use constant from Vec?

Reply:
d5106d4